IQSS/dataverse-frontend
# 136 - Files table UI [2/2] - Row selection

## What this PR does / why we need it:
This PR adds the row selection feature to the UI of the files table. Since the library we're using already has a row selection feature, in this PR I only had to add a few checks to show 2 types of messages related with the row selection. The message about the selection and the message about the zip download limit exceeded. 

This is only the UI no backend connected yet

In this other issue we'll connect the row selection with the download button:

- #63 

## Which issue(s) this PR closes:

- Closes #136 

## Special notes for your reviewer:
Since the library we're using already has a row selection feature, in this PR I only had to add 2 types of messages related with the row selection. The message about the selection and the message about the zip download limit exceeded

Also this is the first time we're using settings so I created a SettingsContext to get the backend settings from the API from any react component

This is only the UI no backend connected yet

## Suggestions on how to test this:
1. `npm i`
2. `cd packages/design-system && npm run build`
3. cd back to the root 
4. `npm run storybook`
5. Check the files table at localhost:6006 storybook in the Dataset page
6. Test the row selection going through the pages selecting different rows
7. Select a few rows and check that a message appears informing about the selection with 2 buttons to select All or to clear selection, ensure this 2 buttons work
8. Select a few rows with a total file size greater than 500 bytes (mocked value) to see the message about the zip download limit

To check the steps in video you can go to https://www.loom.com/share/e970a9abbe6841b59c2e18937532319e?sid=6db11224-ef41-4988-938b-5cc8ce0a57d2

Done!

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/6491efaf-7d2e-4d8c-9720-3582dd764216)

## Is there a release notes update needed for this change?:
Row selection added to the Files Table from the Dataset page

## Additional documentation:
